src/megaapi_impl.cpp

```diff
@@ -7336,7 +7345,7 @@ void MegaApiImpl::logout_result(error e)
     MegaRequestPrivate* request = requestMap.at(client->restag);
     if(!request || (request->getType() != MegaRequest::TYPE_LOGOUT)) return;
 
-    if(!e)
+    if(!e || e == API_ESID)
```

Comment:
I have some doutbs in this case, but I think that it's not possible to receive API_ESID here.